src/internal/operations/backup_test.go

```diff
@@ -31,7 +29,272 @@ import (
 )
 
 // ---------------------------------------------------------------------------
-// unit
+// mocks
+// ---------------------------------------------------------------------------
+
+// ----- restore producer
+
+type mockRestorer struct {
+	gotPaths []path.Path
+}
+
+func (mr *mockRestorer) RestoreMultipleItems(
+	ctx context.Context,
+	snapshotID string,
+	paths []path.Path,
+	bc kopia.ByteCounter,
+) ([]data.Collection, error) {
+	mr.gotPaths = append(mr.gotPaths, paths...)
+
+	return nil, nil
+}
```

Comment:
There are several different structures being used for abstractions. A `helper_test.go` file or `mock_test.go` file would help break up this file.

Reply:
We can take care of that in a future cleanup.